php-checks/src/test/resources/checks/regex/RegexLookaheadCheck.php

```diff
@@ -0,0 +1,26 @@
+<?php
+
+namespace checks\regex;
+
+class RegexLookaheadCheck
+{
+  function noncompliant($input)
+  {
+    preg_match("/(?=a)b/", $input); // Noncompliant {{Remove or fix this lookahead assertion that can never be true.}}
+    preg_match("/(?=ac)ab/", $input); // Noncompliant
+    preg_match("/(?=a)bc/", $input); // Noncompliant
+    preg_match("/(?!a)a/", $input); // Noncompliant
+    preg_match("/(?!ab)ab/", $input); // Noncompliant
```

Comment:
All Noncompliant examples are very simple and don't show that the implementation can detect trickier issues, for example:
```
    preg_match("/(?=a)[^ba]/", $input);
    preg_match("/(?!.)ab/", $input);
```